needy/caches/directory.py

```diff
@@ -31,8 +30,14 @@ def store_file(self, source, key, timeout=0):
         if not os.path.exists(source):
             raise SourceNotFound(source)
         try:
-            clean_file(destination_file)
-            shutil.copyfile(source, destination_file)
```

Comment:
Windows locking is a bit more restrictive. We can't rely on being able to open multiple handles to a file that we have a lock on. So when the cache needs to read or write to a locked file, it needs to use the handle that it has the lock on.